src/plugins/KartesianProfile.cpp

```diff
@@ -195,21 +258,69 @@ void KartesianProfile::endStep(ParticleContainer *particleContainer, DomainDecom
                 _profiles[i]->reset(uID);
             }
         }
-        accumulatedDatasets = 0;
+        _accumulatedDatasets = 0;
     }
 }
 
-unsigned long KartesianProfile::getUID(ParticleIterator thismol) {
-    auto xun = (unsigned) floor(thismol->r(0) * this->universalInvProfileUnit[0]);
-    auto yun = (unsigned) floor(thismol->r(1) * this->universalInvProfileUnit[1]);
-    auto zun = (unsigned) floor(thismol->r(2) * this->universalInvProfileUnit[2]);
-    auto uID = (unsigned long) (xun * this->universalProfileUnit[1] * this->universalProfileUnit[2]
-                           + yun * this->universalProfileUnit[2] + zun);
+unsigned long KartesianProfile::getUID(ParticleIterator &thismol) {
+    auto xun = (unsigned) floor(thismol->r(0) * samplInfo.universalInvProfileUnit[0]);
+    auto yun = (unsigned) floor(thismol->r(1) * samplInfo.universalInvProfileUnit[1]);
+    auto zun = (unsigned) floor(thismol->r(2) * samplInfo.universalInvProfileUnit[2]);
+    auto uID = (unsigned long) (xun * samplInfo.universalProfileUnit[1] * samplInfo.universalProfileUnit[2]
+                           + yun * samplInfo.universalProfileUnit[2] + zun);
     return uID;
 }
 
+long KartesianProfile::getCylUID(ParticleIterator &thismol) {
+

```

Comment:
Add some comments, please, on the R versus R^2 sampling.

Reply:
added comments bbc4c40